File: Tatelier.Nucleus/InputControlItem.cpp
#include "InputControlItem.h"

#include "Input.h"

int InputControlItem::GetCount(int keyCode)
{
	if (!this->IsEnabled())
		return 0;
	return input->GetCount(keyCode);
}

bool InputControlItem::GetKey(int keyCode)
{
	if (!this->IsEnabled())
		return false;
	return input->GetKey(keyCode);
}

bool InputControlItem::GetKeyDown(int keyCode)
{
	if (!this->IsEnabled())
		return false;
	return input->GetKeyDown(keyCode);
}

bool InputControlItem::GetKeyUp(int keyCode)
{
	if (!this->IsEnabled())
		return false;
	return input->GetKeyUp(keyCode);
}

bool InputControlItem::IsEnabled()
{
	return enabled;
}

void InputControlItem::SetEnbaled(bool value)
{
	enabled = value;
}

InputControlItem::InputControlItem()
{
	input = &Input::GetInstance();
}

File: Tatelier/InputControlItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tatelier
{
	class InputControlItem : IInputControlItem
	{
		bool enabled;
		public bool Enabled
        {
            get
            {
				return enabled && InputControl.Singleton.Enabled;
            }
            set
            {
				enabled = value;
            }
        }
		public bool GetKey(int key) => Enabled ? Input.Singleton.GetKey(key) : false;
		public bool GetKeyUp(int key) => Enabled ? Input.Singleton.GetKeyUp(key) : false;
		public bool GetKeyDown(int key) => Enabled ? Input.Singleton.GetKeyDown(key) : false;

		public int GetCount(int key) => Enabled ? Input.Singleton.GetCount(key) : 0;
	}
}
